### PR TITLE
Add Dropdown documentation page

### PR DESCRIPTION
**Pull request by farazcsk**

### CODE REVIEW

File: styleguide/screens/Patterns/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  render() {
+    const { showDropdown } = this.state;
+
+    return (
+      <div>
+        <H level={ 1 }>Dropdown</H>
+        <T elm="p" className={ cx(m.mtr, m.largeI, m.demi) }>
+          Dropdowns are toggleable overlays. They’re typically used to add additional information or
+          functionality to a given element. Only trigger the display of a dropdown using
+          a <C>click</C> or <C>hover</C>.
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Dropdown/Dropdown.js]: Click and hover don't need to be code snippets and I think we should drop "or hover" as it provides a suboptimal experience for touch device users

File: styleguide/screens/Patterns/Dropdown/Dropdown.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        </T>
+        <D>
+          <T elm="p" className={ m.mtr }>
+            For consistency’s sake, all dropdowns also wrap their content in <C>DropdownInner</C>
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Dropdown/Dropdown.js]: > their content in the `DropdownInner` component

?

File: styleguide/screens/Patterns/Dropdown/Dropdown.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                <Target
+                  onClick={ this.handleClick }
+                  placeholderText={ !showDropdown ? 'Open dropdown' : 'Close dropdown' }
+                />
+              }
+              flushHorizontal
+              active={ showDropdown }
+              targetClassName={ css.target }
+            >
+              <Placeholder onClick={ this.handleClick }>Close dropdown</Placeholder>
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Dropdown/Dropdown.js]: A place holder in the dropdown looks really weird. I think we should reserve the use of it specifically for being the top level component in specimen's child prop.

Noticed a bug in the dropdown component too, where you can close it by clicking outside the component yet it doesn't update the state. We should probably look at fixing that!

**Comment by farazcsk** [comment]
@rdjpalmer Good shout! Yeah I think that is sensible.

**Pull request merged**